shared/src/main/scala/tastyquery/ast/Flags.scala

```diff
@@ -0,0 +1,21 @@
+package tastyquery.ast
+
+object Flags:
+
+  object inner:
+    opaque type FlagSet = Long
+    opaque type Flag <: FlagSet = Long
```

Comment:
not really a reason yet, but could perhaps prevent catastrophe in the future if we overload the meaning of some operator like `|` and forget to convert between bit-string and flags

Reply:
As long as we give an explicit type to all the public methods, no catastrophe can happen. And we should always have explicit result types for all public methods anyway.

Reply:
my comment was talking about if in an internal method in the implementation of flags we forget to do the relevant conversion between flags and bitstring - in this case an explicit return type would not give any protection - so I make the inner object for this reason

Reply:
Hum, what do you call a bitstring? The `Long`? Isn't the whole point that `FlagSet`s *are* `Long`s for the purposes of this implementation, so who cares if we forget a conversion here or there? They're the same thing. If we do overload the meaning of an operator such as `|`, it would only ever change the types; not actually change the *meaning* as interpreted on bit sets (because that would be *super* confusing and hence terrible API design).

Reply:
simplified flags implementation